File: js/igboNames.js

function nameGen(type){
	var nm1 = ["Afamefula","Afamefuna","Agu","Akabueze","Akachi","Akubundu","Akuchi","Akumjeli","Amadi","Amazu","Azubuike","Belonwu","Beluchi","Buchi","Chee","Chi","Chibueze","Chibuike","Chibuzo","Chidea","Chidee","Chidey","Chidi","Chidie","Chidiebere","Chidiebube","Chidiegwu","Chidike","Chidubem","Chidy","Chiemeka","Chijindum","Chijioke","Chika","Chike","Chikelu","Chikere","Chima","Chimaijem","Chinasa","Chinaza","Chinedu","Chinelo","Chinese","Chineze","Chino","Chinou","Chinua","Chinuah","Chinwe","Chinweike","Chinwendu","Chinweuba","Chioke","Chioma","Chizoba","Chononso","Chuks","Chukwubuikem","Chukwudi","Chukwuemeka","Chukwuma","Chydea","Chydee","Chydey","Chydi","Chydie","Chydy","Chyke","Diji","Dike","Ebele","Ekene","Ekenedilichukwu","Ekwueme","Emeka","Enyinnaya","Fumnanya","Halim","Halimnye","Ibeabuchi","Ibezimako","Ifeanyichukwu","Ifeatu","Ijeawele","Ijendu","Ikem","Ikenna","Iweobiegbulam","Izuchukwu","Jaja","Jajah","Jamuike","Jel","Jelan","Jelanea","Jelanee","Jelaney","Jelani","Jelanie","Jelany","Jideofor","Jioke","Koofrey","Madou","Madu","Maduabuchim","Madue","Madukaife","Madukwe","Mazi","Mazzi","Munachiso","Ndidi","Ndulu","Ngozi","Nkechi","Nkemdilim","Nnamdi","Nwabudike","Nwankwo","Nweke","Nwoye","Obasea","Obasee","Obasey","Obasi","Obasie","Obasy","Obea","Obee","Obey","Obi","Obie","Obike","Oby","Obyke","Odion","Ogbonna","Ogbonnia","Okafo","Okechukwu","Okeke","Okeli","Okonkwo","Okorie","Okpara","Okparo","Okparra","Okparro","Oluchi","Onuchukwu","Onyekachi","Onyekachukwu","Orjea","Orjee","Orjey","Orji","Orjie","Orjy","Ozioma","Ozoemena","Somayina","Tobechukwu","Uche","Uchea","Uchechea","Uchechee","Uchechey","Uchechi","Uchechie","Uchechy","Uchee","Uchey","Uchi","Uchie","Uchy","Udegbulam","Udegbunam","Udo","Ugochukwu","Ugonna","Ugoorji","Uwaezuoke","Uzochi","Uzochukwu","Uzoma","Yobachukwu","Yobanna","Zebenjo","Zikoranaudodimma"];
	var nm2 = ["Adaeze","Adaezennaya","Adamma","Adanma","Adanna","Adannaya","Adaobi","Adaoma","Adaora","Akachi","Akuchi","Akumjeli","Amachea","Amacheah","Amachee","Amachey","Amachi","Amachie","Amachy","Amachye","Amaka","Amaogechukwu","Amara","Amarachi","Amarachukwu","Anuli","Anulika","Chi","Chiamaka","Chibueze","Chibuifem","Chibuike","Chibuzo","Chichi","Chidea","Chideah","Chidee","Chidey","Chidi","Chidie","Chidiebere","Chidiebube","Chidiegwu","Chidimma","Chidinma","Chidy","Chijindum","Chika","Chikanso","Chike","Chikelu","Chikere","Chimaijem","Chinagorom","Chinaka","Chinasa","Chinaza","Chinecherem","Chinedu","Chineze","Chinonso","Chinwe","Chinweike","Chinwendu","Chinweuba","Chinyelu","Chinyere","Chioma","Chizoba","Chukwudi","Chukwuebuka","Daluchi","Daluchineke","Ebele","Ebuka","Echerem","Ekene","Ekenedilichukwu","Eshe","Fumnanya","Funanya","Gorom","Halim","Halimnye","Ifama","Ifamah","Ifamma","Ifammah","Ifenyinwa","Ifeoma","Ifunanya","Ijendu","Iruka","Iweobiegbulam","Kambiri","Maduabuchim","Madukaego","Mgbankwo","Mgbeke","Mgborie","Munachiso","Ndidi","Ngozi","Nkechi","Nkechinyere","Nkemdilim","Nkiru","Nkiruka","Nneka","Nnenna","Nnenne","Nwanneka","Odera","Ogechi","Ogechukwukama","Olisa","Oluchi","Onochie","Onuchukwu","Onyeka","Onyekachi","Onyekachukwu","Otuosoro","Ozioma","Uchena","Uchenah","Uchenna","Uchennah","Udo","Ula","Urenna","Urunwa","Uzoamaka","Uzochi","Uzochukwu","Uzoma","Yobachukwu","Yobanna","Zikoranaudodimma"];
	var nm3 = ["Akachi","Akuchi","Akumjeli","Chi","Chibueze","Chibuike","Chibuzo","Chidi","Chidiebere","Chidiebube","Chidiegwu","Chijindum","Chika","Chike","Chikelu","Chikere","Chinasa","Chinaza","Chinedu","Chinwe","Chinweike","Chinwendu","Chinweuba","Chioma","Chizoba","Chononso","Ebele","Ekene","Ekenedilichukwu","Fumnanya","Halim","Halimnye","Ijendu","Iweobiegbulam","Maduabuchim","Munachiso","Ndidi","Ngozi","Nkechi","Nkemdilim","Oluchi","Onuchukwu","Onyekachi","Onyekachukwu","Ozioma","Udo","Uzochi","Uzochukwu","Uzoma","Yobachukwu","Yobanna","Zikoranaudodimma"];

	var br = "";

	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd2 = Math.floor(Math.random() * nm1.length);
		if(tp === 1){
			rnd = Math.floor(Math.random() * nm2.length);
			names = nm2[rnd] + " " + nm1[rnd2];
			nm2.splice(rnd, 1);
		}else if(tp === 2){
			rnd = Math.floor(Math.random() * nm3.length);
			names = nm3[rnd] + " " + nm1[rnd2];
			nm3.splice(rnd, 1);
		}else{
			rnd = Math.floor(Math.random() * nm1.length);
			names = nm1[rnd] + " " + nm1[rnd2];
			nm1.splice(rnd, 1);
		}
		nm1.splice(rnd2, 1);
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/igboNames.py
__all__ = ['igboNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nameGen'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['nm1', 'type', 'nm3', 'tp', 'nm2', 'result'])
    var.put('nm1', Js([Js('Afamefula'), Js('Afamefuna'), Js('Agu'), Js('Akabueze'), Js('Akachi'), Js('Akubundu'), Js('Akuchi'), Js('Akumjeli'), Js('Amadi'), Js('Amazu'), Js('Azubuike'), Js('Belonwu'), Js('Beluchi'), Js('Buchi'), Js('Chee'), Js('Chi'), Js('Chibueze'), Js('Chibuike'), Js('Chibuzo'), Js('Chidea'), Js('Chidee'), Js('Chidey'), Js('Chidi'), Js('Chidie'), Js('Chidiebere'), Js('Chidiebube'), Js('Chidiegwu'), Js('Chidike'), Js('Chidubem'), Js('Chidy'), Js('Chiemeka'), Js('Chijindum'), Js('Chijioke'), Js('Chika'), Js('Chike'), Js('Chikelu'), Js('Chikere'), Js('Chima'), Js('Chimaijem'), Js('Chinasa'), Js('Chinaza'), Js('Chinedu'), Js('Chinelo'), Js('Chinese'), Js('Chineze'), Js('Chino'), Js('Chinou'), Js('Chinua'), Js('Chinuah'), Js('Chinwe'), Js('Chinweike'), Js('Chinwendu'), Js('Chinweuba'), Js('Chioke'), Js('Chioma'), Js('Chizoba'), Js('Chononso'), Js('Chuks'), Js('Chukwubuikem'), Js('Chukwudi'), Js('Chukwuemeka'), Js('Chukwuma'), Js('Chydea'), Js('Chydee'), Js('Chydey'), Js('Chydi'), Js('Chydie'), Js('Chydy'), Js('Chyke'), Js('Diji'), Js('Dike'), Js('Ebele'), Js('Ekene'), Js('Ekenedilichukwu'), Js('Ekwueme'), Js('Emeka'), Js('Enyinnaya'), Js('Fumnanya'), Js('Halim'), Js('Halimnye'), Js('Ibeabuchi'), Js('Ibezimako'), Js('Ifeanyichukwu'), Js('Ifeatu'), Js('Ijeawele'), Js('Ijendu'), Js('Ikem'), Js('Ikenna'), Js('Iweobiegbulam'), Js('Izuchukwu'), Js('Jaja'), Js('Jajah'), Js('Jamuike'), Js('Jel'), Js('Jelan'), Js('Jelanea'), Js('Jelanee'), Js('Jelaney'), Js('Jelani'), Js('Jelanie'), Js('Jelany'), Js('Jideofor'), Js('Jioke'), Js('Koofrey'), Js('Madou'), Js('Madu'), Js('Maduabuchim'), Js('Madue'), Js('Madukaife'), Js('Madukwe'), Js('Mazi'), Js('Mazzi'), Js('Munachiso'), Js('Ndidi'), Js('Ndulu'), Js('Ngozi'), Js('Nkechi'), Js('Nkemdilim'), Js('Nnamdi'), Js('Nwabudike'), Js('Nwankwo'), Js('Nweke'), Js('Nwoye'), Js('Obasea'), Js('Obasee'), Js('Obasey'), Js('Obasi'), Js('Obasie'), Js('Obasy'), Js('Obea'), Js('Obee'), Js('Obey'), Js('Obi'), Js('Obie'), Js('Obike'), Js('Oby'), Js('Obyke'), Js('Odion'), Js('Ogbonna'), Js('Ogbonnia'), Js('Okafo'), Js('Okechukwu'), Js('Okeke'), Js('Okeli'), Js('Okonkwo'), Js('Okorie'), Js('Okpara'), Js('Okparo'), Js('Okparra'), Js('Okparro'), Js('Oluchi'), Js('Onuchukwu'), Js('Onyekachi'), Js('Onyekachukwu'), Js('Orjea'), Js('Orjee'), Js('Orjey'), Js('Orji'), Js('Orjie'), Js('Orjy'), Js('Ozioma'), Js('Ozoemena'), Js('Somayina'), Js('Tobechukwu'), Js('Uche'), Js('Uchea'), Js('Uchechea'), Js('Uchechee'), Js('Uchechey'), Js('Uchechi'), Js('Uchechie'), Js('Uchechy'), Js('Uchee'), Js('Uchey'), Js('Uchi'), Js('Uchie'), Js('Uchy'), Js('Udegbulam'), Js('Udegbunam'), Js('Udo'), Js('Ugochukwu'), Js('Ugonna'), Js('Ugoorji'), Js('Uwaezuoke'), Js('Uzochi'), Js('Uzochukwu'), Js('Uzoma'), Js('Yobachukwu'), Js('Yobanna'), Js('Zebenjo'), Js('Zikoranaudodimma')]))
    var.put('nm2', Js([Js('Adaeze'), Js('Adaezennaya'), Js('Adamma'), Js('Adanma'), Js('Adanna'), Js('Adannaya'), Js('Adaobi'), Js('Adaoma'), Js('Adaora'), Js('Akachi'), Js('Akuchi'), Js('Akumjeli'), Js('Amachea'), Js('Amacheah'), Js('Amachee'), Js('Amachey'), Js('Amachi'), Js('Amachie'), Js('Amachy'), Js('Amachye'), Js('Amaka'), Js('Amaogechukwu'), Js('Amara'), Js('Amarachi'), Js('Amarachukwu'), Js('Anuli'), Js('Anulika'), Js('Chi'), Js('Chiamaka'), Js('Chibueze'), Js('Chibuifem'), Js('Chibuike'), Js('Chibuzo'), Js('Chichi'), Js('Chidea'), Js('Chideah'), Js('Chidee'), Js('Chidey'), Js('Chidi'), Js('Chidie'), Js('Chidiebere'), Js('Chidiebube'), Js('Chidiegwu'), Js('Chidimma'), Js('Chidinma'), Js('Chidy'), Js('Chijindum'), Js('Chika'), Js('Chikanso'), Js('Chike'), Js('Chikelu'), Js('Chikere'), Js('Chimaijem'), Js('Chinagorom'), Js('Chinaka'), Js('Chinasa'), Js('Chinaza'), Js('Chinecherem'), Js('Chinedu'), Js('Chineze'), Js('Chinonso'), Js('Chinwe'), Js('Chinweike'), Js('Chinwendu'), Js('Chinweuba'), Js('Chinyelu'), Js('Chinyere'), Js('Chioma'), Js('Chizoba'), Js('Chukwudi'), Js('Chukwuebuka'), Js('Daluchi'), Js('Daluchineke'), Js('Ebele'), Js('Ebuka'), Js('Echerem'), Js('Ekene'), Js('Ekenedilichukwu'), Js('Eshe'), Js('Fumnanya'), Js('Funanya'), Js('Gorom'), Js('Halim'), Js('Halimnye'), Js('Ifama'), Js('Ifamah'), Js('Ifamma'), Js('Ifammah'), Js('Ifenyinwa'), Js('Ifeoma'), Js('Ifunanya'), Js('Ijendu'), Js('Iruka'), Js('Iweobiegbulam'), Js('Kambiri'), Js('Maduabuchim'), Js('Madukaego'), Js('Mgbankwo'), Js('Mgbeke'), Js('Mgborie'), Js('Munachiso'), Js('Ndidi'), Js('Ngozi'), Js('Nkechi'), Js('Nkechinyere'), Js('Nkemdilim'), Js('Nkiru'), Js('Nkiruka'), Js('Nneka'), Js('Nnenna'), Js('Nnenne'), Js('Nwanneka'), Js('Odera'), Js('Ogechi'), Js('Ogechukwukama'), Js('Olisa'), Js('Oluchi'), Js('Onochie'), Js('Onuchukwu'), Js('Onyeka'), Js('Onyekachi'), Js('Onyekachukwu'), Js('Otuosoro'), Js('Ozioma'), Js('Uchena'), Js('Uchenah'), Js('Uchenna'), Js('Uchennah'), Js('Udo'), Js('Ula'), Js('Urenna'), Js('Urunwa'), Js('Uzoamaka'), Js('Uzochi'), Js('Uzochukwu'), Js('Uzoma'), Js('Yobachukwu'), Js('Yobanna'), Js('Zikoranaudodimma')]))
    var.put('nm3', Js([Js('Akachi'), Js('Akuchi'), Js('Akumjeli'), Js('Chi'), Js('Chibueze'), Js('Chibuike'), Js('Chibuzo'), Js('Chidi'), Js('Chidiebere'), Js('Chidiebube'), Js('Chidiegwu'), Js('Chijindum'), Js('Chika'), Js('Chike'), Js('Chikelu'), Js('Chikere'), Js('Chinasa'), Js('Chinaza'), Js('Chinedu'), Js('Chinwe'), Js('Chinweike'), Js('Chinwendu'), Js('Chinweuba'), Js('Chioma'), Js('Chizoba'), Js('Chononso'), Js('Ebele'), Js('Ekene'), Js('Ekenedilichukwu'), Js('Fumnanya'), Js('Halim'), Js('Halimnye'), Js('Ijendu'), Js('Iweobiegbulam'), Js('Maduabuchim'), Js('Munachiso'), Js('Ndidi'), Js('Ngozi'), Js('Nkechi'), Js('Nkemdilim'), Js('Oluchi'), Js('Onuchukwu'), Js('Onyekachi'), Js('Onyekachukwu'), Js('Ozioma'), Js('Udo'), Js('Uzochi'), Js('Uzochukwu'), Js('Uzoma'), Js('Yobachukwu'), Js('Yobanna'), Js('Zikoranaudodimma')]))
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('names', ((var.get('nm2').get(var.get('rnd'))+Js(' '))+var.get('nm1').get(var.get('rnd2'))))
                var.get('nm2').callprop('splice', var.get('rnd'), Js(1.0))
            else:
                if PyJsStrictEq(var.get('tp'),Js(2.0)):
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                    var.put('names', ((var.get('nm3').get(var.get('rnd'))+Js(' '))+var.get('nm1').get(var.get('rnd2'))))
                    var.get('nm3').callprop('splice', var.get('rnd'), Js(1.0))
                else:
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                    var.put('names', ((var.get('nm1').get(var.get('rnd'))+Js(' '))+var.get('nm1').get(var.get('rnd2'))))
                    var.get('nm1').callprop('splice', var.get('rnd'), Js(1.0))
            var.get('nm1').callprop('splice', var.get('rnd2'), Js(1.0))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
pass
pass


# Add lib to the module scope
igboNames = var.to_python()